notification/src/handler/notification/notification.handler.js

```diff
@@ -42,10 +42,23 @@ async function processNotification(
     return
   }
 
+  const pspReference = _.get(
```

Comment:
This `pspReference` is useless for us at this moment and you don't need it. It's a completely new ID for this notification (= ID of the transaction). See my comment in the JIRA task where I explain the meanings of reference fields in the notification.

Reply:
That is right, but here the new pspReference is being used to search for the payment by key. That's why it is useless because this pspReference is not saved anywhere yet and cannot be used here to search for the payment by key.

Reply:
It's correct for web component before  v5.0 as we can get `pspReference` and set payment key from extension module. But for web componenet v5 we can only get `pspReference` via notification. We implement this now in notification because it can apply on web component both before and after v5.

Reply:
@leungkinghin but what's your intention with `pspReference` here? Currently you use the `pspReference` to search for payment, right? When this is a valid use-case?

Reply:
According to current change, `pspReference` is set to key only after payment is authorized. So in other operations like refund, capture, cancel, etc. Actually we can lookup payment in notification by merchantReference and original reference in refund/capture/cancel notification. Is it your meaning?

If so, i think it has dependency on another comment. Consider the use case that two authorization notification come in a row, the first one is failure and the second one is succeed. `pspReference` would be needed here if we apply the change for the comment below
https://github.com/commercetools/commercetools-adyen-integration/pull/1049/commits/f55dfd2b4239074f4bdcaa656fe068e2e2cf818b#r1056408788

Reply:
Hmmm that's a weird use-case and I'm not sure if it's even a valid use case 🤔 Please clarify your statement with Adyen.

Reply:
I tried to remove `pspReference` and put `paymentKey = originalReference || merchantReference`. However I have to undo many changes in test cases, since we don't need to get payment by multiple keys anymore. I would rather work on it after holiday.

Reply:
@lojzatran https://github.com/commercetools/commercetools-adyen-integration/pull/1049/commits/66e60fec7021f7804302dce8c541972557fac4e6

Reply:
reverted, because payment key has been updated already in extension module after making payment. so that `pspReference` is still required to look up the payment in authorization notification, in which no original reference has provided yet.